### PR TITLE
Create notification channels

### PR DESCRIPTION
**Pull request by paulnabanita1**
Implementing notification channels for Android versions Oreo and above.
This code will be maintaining backward compatibility with older versions of Android.
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `gradlew connectedObaGoogleDebugAndroidTest connectedObaAmazonDebugAndroidTest` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them for the initial submission of the pull request.  When addressing comments on a pull request, please push a new commit per comment when possible (reviewers will squash and merge using GitHub merge tool)

### CODE REVIEW

**Comment by barbeau** [comment]
Thanks @paulnabanita1!

Did you test the arrival reminder with a recurring reminder across multiple days (vs a one time reminder)?

If not, could you please do this? 

We're seeing issues with background crashes for a service running in the background for the EmbeddedSocial SDK and I want to make sure we don't have the same issue with the service that runs prior to the bus arriving to check arrivals times to see if the reminder notification should be triggered.

**Comment by paulnabanita1** [comment]
Thank you @barbeau for the comments.
I haven't tested the change for a recurring reminder across multiple days. I will test it and let you know if all working or not.

**Comment by barbeau** [comment]
@paulnabanita1 Actually, to test this you'll need https://github.com/OneBusAway/onebusaway-android/pull/930 too.  Could you rebase this pull request on top of https://github.com/OneBusAway/onebusaway-android/pull/930, and then test?

https://github.com/OneBusAway/onebusaway-android/pull/930 is nearly done, I just have to implement runtime permissions for data backup/restore.

**Comment by paulnabanita1** [comment]
@barbeau Sure! I will rebase my PR and then test it.

File: onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java [hunk 1]
```diff
@@ -70,6 +73,9 @@
     // Region preference (long id)
     private static final String TAG = "Application";
 
+    public static final String CHANNEL_REALTIME_ID = "ch101";
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java]: Could you please change the name of `CHANNEL_REALTIME_ID` to `CHANNEL_TRIP_PLAN_UPDATES_ID`?

File: onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java [hunk 2]
```diff
@@ -70,6 +73,9 @@
...
+    public static final String CHANNEL_BUS_ARRIVAL_ID = "ch102";
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java]: Could you please change the name of `CHANNEL_BUS_ARRIVAL_ID` to `CHANNEL_ARRIVAL_REMINDERS_ID`?

File: onebusaway-android/src/main/java/org/onebusaway/android/directions/realtime/RealtimeService.java [hunk 3]
```diff
@@ -243,7 +243,7 @@ private void showNotification(ItineraryDescription description, int title, int m
                         PendingIntent.FLAG_CANCEL_CURRENT);
 
         NotificationCompat.Builder mBuilder =
-                new NotificationCompat.Builder(getApplicationContext())
+                new NotificationCompat.Builder(getApplicationContext(),Application.CHANNEL_REALTIME_ID)
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/directions/realtime/RealtimeService.java]: Could you please change `,Application.CHANNEL_REALTIME_ID` to `, Application.CHANNEL_REALTIME_ID` (add a space)?

File: onebusaway-android/src/main/java/org/onebusaway/android/ui/TripResultsFragment.java [hunk 4]
```diff
@@ -333,10 +337,20 @@ void updateInfo() {
...
-            if(Application.getPrefs()
-                    .getBoolean(getString(R.string.preference_key_trip_plan_notifications), true)) {
+            Context context = Application.get().getApplicationContext();
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                NotificationManager manager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+                NotificationChannel channel = manager.getNotificationChannel(Application.CHANNEL_REALTIME_ID);
+                if(channel.getImportance() != NotificationManager.IMPORTANCE_NONE){
+                    RealtimeService.start(getActivity(), getArguments());
+                }
+            }
```
Comment (barbeau) [inline onebusaway-android/src/main/java/org/onebusaway/android/ui/TripResultsFragment.java]: Could you please put the else brackets on same line and add a space?

`} else {`

**Comment by barbeau** [comment]
@paulnabanita1 It looks like a bunch of other commits were accidentally added to this PR.  Could you please take a look?  If you need help with this please let me know.

**Comment by barbeau** [comment]
To be replaced by https://github.com/OneBusAway/onebusaway-android/pull/935

**Pull request closed without merging**